[Draft] Refactor voltage regulation in iidm (#3699)
powsybl/powsybl-core: docs/grid_model/additional.md

| $TargetValue$    | kV or MVar  | The voltage target or the reactive target at regulating terminal                                                                       |
| $TargetDeadband$ | kV          | The deadband used to avoid excessive update of controls (`RatioTapChanger` and `ShuntCompensator`)                                     |
| $Slope$          | kV per MVar | The sensibility of the voltage with respect to reactive power (`VOLTAGE_PER_REACTIVE_POWER` or `REACTIVE_POWER_PER_ACTIVE_POWER` mode) |
| $Terminal$       |             | The regulating Terminal which can be remote or local                                                                                   |
| $Mode$           |             | The kind of regulation                                                                                                                 |
| $Regulating$     |             | True if the equipment is regulating, false otherwise                                                                                   |

**Specifications**

> The values `Regulating` and `Mode` are required and `TargetValue` is required if the `Terminal` is set

Reviewer: ```suggestion
The values `Regulating` and `Mode` are always required. `TargetValue` is required when the `Terminal` is set.
```

`Terminal` is optional. If not set, the local terminal of the connectable will be used.  
Once a `Terminal` has been set, it is considered a `Remote` terminal even if it is the connectable Terminal (local).  

> `TargetValue` is used only if the `Terminal` is set. Otherwise, the local target voltage or reactive value is used

Reviewer: ```suggestion
`TargetValue` is used only when the `Terminal` is set. Otherwise, the local target voltage or reactive value is used.
```

| $TargetValue$    | kV or MVar  | The voltage target or the reactive target at regulating terminal                                                                       |
| $TargetDeadband$ | kV          | The deadband used to avoid excessive update of controls (`RatioTapChanger` and `ShuntCompensator`)                                     |
| $Slope$          | kV per MVar | The sensibility of the voltage with respect to reactive power (`VOLTAGE_PER_REACTIVE_POWER` or `REACTIVE_POWER_PER_ACTIVE_POWER` mode) |
| $Terminal$       |             | The regulating Terminal which can be remote or local                                                                                   |
| $Mode$           |             | The kind of regulation                                                                                                                 |
| $Regulating$     |             | True if the equipment is regulating, false otherwise                                                                                   |

**Specifications**

> TODO MSA add information when the voltageRegulation is absent

Reviewer: TODO MSA to handle

Unlike in CGMES, the VoltageRegulation object isn't shared: each IIDM object capable of voltage regulation have its own.

Here the list of objects capable of such regulation by authorized mode:

| Equipment                                                                  | Voltage | Reactive Power | Voltage Per Reactive Power | Reactive Power Per Active Power |
|----------------------------------------------------------------------------|---------|----------------|----------------------------|---------------------------------|
| [Battery](./network_subnetwork.md#battery)                                 | X       | X (*)          |                            |                                 |
| [Generator](./network_subnetwork.md#generator)                             | X       | X (*)          |                            | X                               |
| [RatioTapChanger](#ratio-tap-changer)                                      | X (*)   | X (*)          |                            |                                 |

> | [ShuntCompensator](./network_subnetwork.md#shunt-compensator)              | X       |                |                            |                                 |

Reviewer 2: I think extension documentation related to static VAR compensator needs to be updated

https://github.com/powsybl/powsybl-core/blob/0b29d5020f5df3c96b4b98f2a9d0e2219617e5d0/docs/grid_model/extensions.md?plain=1#L990-L993

Author: Yes, indeed
The Remote reactive power control extension needs updating
There is nothing to do regarding the Voltage Regulation extension as it is missing

Reviewer 2: Yes, as the extension is missing, I think it can be removed from the doc also

`Terminal` is optional. If not set, the local terminal of the connectable will be used.  
Once a `Terminal` has been set, it is considered a `Remote` terminal even if it is the connectable Terminal (local).  
`TargetValue` is used only when the `Terminal` is set. Otherwise, the local target voltage or reactive value is used.

Regulation `Mode` has the following values : 
- `VOLTAGE`
- `REACTIVE_POWER`
- `VOLTAGE_PER_REACTIVE_POWER`
- `REACTIVE_POWER_PER_ACTIVE_POWER`

The optional `Slope` attribute is relevant for:
- `VOLTAGE_PER_REACTIVE_POWER`: it corresponds to the $\lambda$ in $U_0 = U + \lambda \times Q$
- `REACTIVE_POWER_PER_ACTIVE_POWER`: it corresponds to the $tan(\phi)$ in $Q = tan(\phi) \times P$

The optional `TargetDeadband` is only pertinent for objects with discrete (as opposed to continuous) voltage regulation, which is the case for [RatioTapChanger](#ratio-tap-changer)  and [ShuntCompensator](./network_subnetwork.md#shunt-compensator)

**Example**

This example shows how to add a voltage regulation to a generator with remote regulation:
 - when we create a new generator:
```java

> generator.newVoltageRegulation()

Reviewer: ```suggestion
generatorAdder.newVoltageRegulation()
```
